### PR TITLE
A64: Implement CLZ

### PR DESCRIPTION
**Pull request by lioncash**

### CODE REVIEW

File: src/backend_x64/emit_x64.cpp [hunk 1]
```diff
@@ -1327,6 +1327,31 @@ void EmitX64<JST>::EmitCountLeadingZeros(EmitContext& ctx, IR::Inst* inst) {
...
+       code->lzcnt(result, source);
+
+       ctx.reg_alloc.DefineValue(inst, result);
+   } else {
+       Xbyak::Reg64 source = ctx.reg_alloc.UseScratchGpr(args[0]).cvt64();
+       Xbyak::Reg64 result = ctx.reg_alloc.ScratchGpr().cvt64();
+
+       // The result of a bsr of zero is undefined, but zf is set after it.
+       code->bsr(result, source);
+       code->mov(source, 0xFFFFFFFFFFFFFFFF);
```
Comment (MerryMage) [inline src/backend_x64/emit_x64.cpp]: Smaller codesize:

    code->mov(source.cvt32(), 0xFFFFFFFF);
    code->cmovz(result.cvt32(), source.cvt32());
    code->neg(result.cvt32());
    code->add(result.cvt32(), 63);

Upper 32 bits of a register are always zeroed after a 32 bit operation on x64.

File: src/frontend/ir/ir_emitter.h [hunk 2]
```diff
@@ -135,7 +135,9 @@ class IREmitter {
     U32 ByteReverseWord(const U32& a);
     U16 ByteReverseHalf(const U16& a);
     U64 ByteReverseDual(const U64& a);
-    U32 CountLeadingZeros(const U32& a);
+    U32 CountLeadingZeros32(const U32& a);
+    U64 CountLeadingZeros64(const U64& a);
```
Comment (MerryMage) [inline src/frontend/ir/ir_emitter.h]: Do you prefer the explicitness of `CountLeadingZeros{32,64}` versus calling them all `CountLeadingZeros` and letting overload resolution sort it out?

Comment (lioncash) [reply src/frontend/ir/ir_emitter.h]: ah, that was a carry-over the emitter function naming, I'll correct that.

File: src/backend_x64/emit_x64.cpp (hunk 1 above)
Comment (lioncash) [reply src/backend_x64/emit_x64.cpp]: Ah right, forgot about that. Good catch.

File: src/frontend/ir/ir_emitter.h (hunk 2 above)
Comment (lioncash) [reply src/frontend/ir/ir_emitter.h]: ah, that was a carry-over the emitter function naming, I'll correct that.

File: src/backend_x64/emit_x64.cpp (hunk 1 above)
Comment (lioncash) [reply src/backend_x64/emit_x64.cpp]: Forgot about that. Good catch.

**Comment by lioncash** [comment]
@MerryMage should be all addressed now

**Pull request merged**